builtins/web/form-data/form-data-encoder.h

```diff
@@ -0,0 +1,45 @@
+#ifndef BUILTINS_WEB_FORM_DATA_ENCODER_
+#define BUILTINS_WEB_FORM_DATA_ENCODER_
+
+#include "builtin.h"
+
+namespace builtins {
+namespace web {
+namespace form_data {
+
+class MultipartFormDataImpl;
+
+class MultipartFormData : public FinalizableBuiltinImpl<MultipartFormData> {
+  static MultipartFormDataImpl *as_impl(JSObject *self);
+
+  static bool read(JSContext *cx, HandleObject self, std::span<uint8_t> buf,
+                   size_t start, size_t *read, bool *done);
+
+public:
+  static constexpr const char *class_name = "MultipartFormData";
+  static constexpr unsigned ctor_length = 0;
+
+  static const JSFunctionSpec static_methods[];
+  static const JSPropertySpec static_properties[];
+  static const JSFunctionSpec methods[];
+  static const JSPropertySpec properties[];
+
+  enum Slots { Form, Inner, Count };
+
+  static JSObject *form_data(JSObject *self);
+  static std::string boundary(JSObject *self);
+
+  static size_t query_length(JSContext *cx, HandleObject self);
+  static JSObject *encode_stream(JSContext *cx, HandleObject self);
+  static JSObject *create(JSContext *cx, HandleObject form_data);
```

Comment:
The implementation of these seems to be missing? (And in the case of `query_length` I also can't find any uses.)

Reply:
The `query_length` function is now implemented. Without setting the `Content-Length` header uploading `FormData` tests were failing.